src/huggingface_hub/hf_api.py

```diff
@@ -4967,6 +4975,227 @@ def snapshot_download(
             tqdm_class=tqdm_class,
         )
 
+    def get_safetensors_metadata(
+        self,
+        repo_id: str,
+        *,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> SafetensorsRepoMetadata:
+        """
+        Parse metadata for a safetensors repo on the Hub.
+
+        We first check if the repo has a single safetensors file or a sharded safetensors repo. If it's a single
+        safetensors file, we parse the metadata from this file. If it's a sharded safetensors repo, we parse the
+        metadata from the index file and then parse the metadata from each shard.
+
+        To parse metadata from a single safetensors file, use [`get_safetensors_metadata`].
+
+        For more details regarding the safetensors format, check out https://huggingface.co/docs/safetensors/index#format.
```

Comment:
you'll be able to hyper-link to this implem from the safetensors doc too BTW (like i had done for the JS implem)

Reply:
more generally let's always make sure we cross-link stuff as much as possible